tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Transients.scala

```diff
@@ -0,0 +1,30 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2018, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.core.tools.linker.backend.emitter
+
+import org.scalajs.core.ir.Trees._
+import org.scalajs.core.ir.Printers._
+
+object Transients {
+
+  final case class CallHelper(helper: String, args: List[Tree])
+      extends Transient.Value {
+
+    def printIR(out: java.io.Writer, irPrinter: IRTreePrinter): Unit = {
+      out.write("$callHelper(")
+      out.write(helper)
+      for (arg <- args) {
+        out.write(", ")
+        irPrinter.print(arg)
```

Comment:
Hmmm... These two lines make the relationship between `irPrinter` and `out` appear magic :-/ Alternative would be to make `IRTreePrinter.print(_: String)` public, but that's also not nice...

Have you considered creating a dedicated type for this? (that has `print(_: String)` and `print(_: Tree)`?)

Reply:
`IRTreePrinter` could implement all the methods of `Writer` by forwarding them to `out`. That would make the relationship clearer.

Just `String` and `Tree` wouldn't be enough. There's also `Type` and several others.